Size t (#3488)
modelica/ModelicaSpecification: chapters/functions.tex

  In case calls to several external functions are generated in the same translation unit, the \lstinline!Include! annotations of the different functions must not define the same function -- except when relying on the deprecated behavior.

  The included code should be valid C89 code.

>   If the code uses arrays the tool is responsible for ensuring that a C-header defining \lstinline[language=C]!size_t! is included before this code.

Reviewer: One shouldn't need to inspect the "code":
```suggestion
  If the \lstinline[language=grammar]!external-function-call! uses arrays, the tool is responsible for ensuring that a C-header defining \lstinline[language=C]!size_t! is included before the \lstinline!"insertedCode"!.
```

>   If the code uses arrays the tool is responsible for ensuring that a C-header defining \lstinline[language=C]!size_t! is included before this code.

Reviewer: One shouldn't need to inspect the "code".  When trying to come up with a precise rule, I can't think of anything better than going for the `size`-expression:
```suggestion
  If the \lstinline[language=grammar]!external-function-call! contains any `size`-expression, the tool is responsible for ensuring that a C-header defining \lstinline[language=C]!size_t! is included before the \lstinline!"insertedCode"!.
```

>   The user is responsible for ensuring that the C-code can be compiled separately -- with \lstinline[language=C]!#include <stddef.h>! prepended if using any array.

Reviewer: We always use the spaced form of _Modelica code_, and should do the same for C:
```suggestion
  The user is responsible for ensuring that the C code can be compiled separately -- with \lstinline[language=C]!#include <stddef.h>! prepended if using any array.
```

Reviewer: Also, there should be no need to spell out that it is the user who is responsible; only the user can be responsible for the annotation:
```suggestion
  The \lstinline!"insertedCode"! (with \lstinline[language=C]!#include <stddef.h>! prepended when this would be a tool responsibility) shall be possible to compile separately.
```

Reviewer: I'm afraid that this formulation gives the impression that it is _stddef.h_ that tools are expected to use, so maybe we should be more careful if we want to avoid code becoming dependent upon this particular header…

Reviewer: Like this:
```suggestion
  The \lstinline!"insertedCode"! (following the inclusion of a header for \lstinline[language=C]!size_t! when this would be a tool responsibility) shall be possible to compile separately.
```

>   The \lstinline!"insertedCode"! (following the inclusion of a header for \lstinline[language=C]!size_t! when this would be a tool responsibility) shall be possible to compile separately.

Author: I agree with the intention of this, but the formulation is now quite complicated. It has conditionals and parts in parenthesis.
Could we simplify it to something like:
```suggestion
  This gives a code fragment \lstinline!"insertedCode"! conditionally preceded by a header for \lstinline[language=C]!size_t!.
  This code fragment shall be possible to compile separately.
```

Author: @henrikt-ma ?

Reviewer: Sure, it would be nice to simplify.  However, I don't find the suggestion clear enough.  Let me try something else:
```suggestion
  With at most the tool-supplied inclusion of a header for \lstinline[language=C]!size_t! prepended, the \lstinline!"insertedCode"! shall be possible to compile separately.
```

Reviewer: Now I also see the intention of your suggestion, namely _a code fragment consisting of…_.  That could work too, but I wouldn't call it a _code fragment_ considering it's something as complete as a translation unit.

Reviewer 2: I prefer Hans's variant with two sentences with a small change to the second sentence.
```suggestion
This gives a code fragment \lstinline!"insertedCode"! conditionally preceded by a header for \lstinline[language=C]!size_t!.
It must be possible to compile this code fragment separately.
```
